django/db/backends/sqlite3/base.py

```diff
@@ -470,3 +497,233 @@ def _sqlite_regexp(re_pattern, re_string):
+
+def _sqlite_cot(x):
+    return -math.tan(3.0*math.pi/2.0 + x)
+
+def _sqlite_degrees(x):
+    return math.degrees(x)
+
+def _sqlite_exp(x):
+    return math.exp(x)
+
+def _sqlite_floor(x):
+    return math.floor(x)
+
+def _sqlite_log(x):
+    return math.log(x)
+
+def _sqlite_mod(x, y):
+    return x%y
+=======
+    res = None
+    try:
+        res = math.acos(x)
+    except TypeError as e1:
+        warnings.warn(e1)
+    except ValueError as e2:
+        warnings.warn(e2)
+    finally:
+        return res
+
+def _sqlite_asin(x):
+    res = None
+    try:
+        res = math.asin(x)
+    except TypeError as e1:
+        warnings.warn(e1)
+    except ValueError as e2:
+        warnings.warn(e2)
+    finally:
+        return res
+
+def _sqlite_atan(x):
+    res = None
+    try:
+        res = math.atan(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+def _sqlite_atan2(x, y):
+    res = None
+    try:
+        res = math.atan2(x, y)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+def _sqlite_ceiling(x):
+    res = None
+    try:
+        res = math.ceil(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+def _sqlite_cos(x):
+    res = None
+    try:
+        res = math.cos(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+def _sqlite_cot(x):
+    res = None
+    if x == 0:
+        return None
+    try:
+        res = -math.tan(3.0*math.pi/2.0 + x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+def _sqlite_degrees(x):
+    res = None
+    try:
+        res = math.degrees(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+def _sqlite_exp(x):
+    res = None
+    try:
+        res = math.exp(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+def _sqlite_floor(x):
+    res = None
+    try:
+        res = math.floor(x)
+    except TypeError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+def _sqlite_log(x):
+    res = None
+    try:
+        res = math.log(x)
+    except TypeError as e:
+        warnings.warn(e)
+    except ValueError as e:
+        warnings.warn(e)
+    finally:
+        return res
+
+def _sqlite_mod(x, y):
+    res = None
+    try:
+        res = x%y
+    except TypeError as e:
+        warnings.warn(e)
+    except ZeroDivisionError as e:
+        warnings.warn(e)
+    finally:
+        return res    
+>>>>>>> ticket_28643
+
+def _sqlite_pi():
+    return math.pi
+
+def _sqlite_power2(x, y):
+<<<<<<< HEAD
+    return x**y
+
+def _sqlite_radians(x):
+    return math.radians(x)
+
+def _sqlite_sin(x):
+    return math.sin(x)
+
+def _sqlite_sqrt(x):
+    return math.sqrt(x)
+
+def _sqlite_tan(x):
+    return math.tan(x)
+
+=======
+    res = None
+    try:
+        res = x**y
+    except TypeError as e:
```

Comment:
I think you could merge into one branch.

Reply:
Hi atombrella,

Thank you so much for the reviews. I will work on it and try to finish it within one week. Sorry that I am a little busy this weekend and early next week.